src/library.js

```diff
@@ -1508,6 +1508,7 @@ LibraryManager.library = {
   get setjmp__deps() {
     return this.longjmp__deps;
   },
+#if DISABLE_EXCEPTION_CATCHING
```

Comment:
What is this for?

Reply:
When exceptions are enabled we can't error out here when `emscripten_longjmp` is needed, because the exception handling code in `std::__terminate` requires `emscripten_longjmp` now.

Reply:
Basically the thing is error is trying to detect is occurring now for all programs that use exception handling.   But maybe that is ok?  I guess the downside would be that some users might not see this error when they otherwise would?   i.e. if they really are trying to use setjmp longjmp in their code but building with SUPPORT_LONGJMP=0 they would previously have gotten a linker here here..

Reply:
Why does `std::__terminate` depend on longjmp now? Might also be worth a source comment, seems surprising.

Reply:
When exception are enabled and longjmp is also enabled the correct behavior of catch(...) is to rethrow any longjmp that is catches.  And std::__terminate contains a `catch(...)` and we always build libcxxabit with SjLj enabled:

https://github.com/emscripten-core/emscripten/blob/f782b50a7f8dded7cd0e2c7ee4fed41ab743f5c0/system/lib/libcxxabi/src/cxa_handlers.cpp#L64-L68

Its not just std::__terminate, libcxx and libcxxabi have a lot catches like this.

I think we going to revert the llvm-side change that introduced this dependency (for now).